In [92]:
from math import factorial as fcl
import numpy as np

Формула расчёта кол-ва комбинаций:
$$C_{n}^{k} = \frac{n!}{k!(n-k)!}$$

In [14]:
def comb(n, k):
    C = fcl(n)/(fcl(k)*fcl(n-k))
    return int(C)

Формула расчёта вероятности:
$$P(A) = \frac{m}{n}$$

In [34]:
def ver(m, n):
    P = m/n
    return float(P)

Формула Байеса:
$$P(B|A)=\frac{P(B) \cdot P(A|B)}{P(A)}$$

### Задание 1.

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки

#### Решение:

In [84]:
salary = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

Среднее арифметическое:

In [86]:
average = sum(salary)/len(salary)
average

65.3

проверка:

In [96]:
np.mean(salary)

65.3

Среднее квадратичное отклонение:

In [94]:
E = 0
for num in salary:
    x = (num - average)**2
    E += x
S = np.sqrt(E/len(salary))
S

30.823854398825596

проверка:

In [95]:
np.std(salary)

30.823854398825596

Смещённая оценка дисперсии:

In [97]:
E = 0
for num in salary:
    x = (num - average)**2
    E += x
D = E/len(salary)
D

950.11

проверка:

In [98]:
np.var(salary)

950.11

Несмещённая оценка дисперсии:

In [100]:
E = 0
for num in salary:
    x = (num - average)**2
    E += x
D_n = E/(len(salary)-1)
D_n

1000.1157894736842

### Задание 2.

 В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

#### Решение:

Для выполнения условия задачи:
из первого ящика можно вытащить или 2, или 1, или 0 белых;
из второгоя ящика можно вытащить или 4, или 3, или 2, или 1 белый

Рассчитаем количество комбинаций белых мячей в 1м ящике из 5 по 2:

In [57]:
comb2_5 = comb(5, 2)
comb1_5= comb(5, 1)
print(f'количество комбинаций 2 белых {comb2} \nколичество комбинаций 1 белого {comb1}')

количество комбинаций 2 белых 10 
количество комбинаций 1 белого 5


Рассчитаем количество комбинаций белых мячей в 1м ящике из 8 по 2:

In [58]:
comb2_8 = comb(8, 2)
comb1_8 = comb(8, 1)
print(f'количество комбинаций 2 белых {comb2_8} \nколичество комбинаций 1 белого {comb1_8}')

количество комбинаций 2 белых 28 
количество комбинаций 1 белого 8


Рассчитаем вероятности для первого ящика:

In [53]:
ver2_1 = ver(comb2_5, comb2_8)
ver1_1 = ver(comb1_5, comb1_8)
ver0_1 = 1 - ver1_1
print(f' вероятность извлечения 2 белых {ver2_1} \n вероятность извлечения 1 белого {ver1_1} \n вероятность извлечения 0 белых {ver0_1}')

 вероятность извлечения 2 белых 0.35714285714285715 
 вероятность извлечения 1 белого 0.625 
 вероятность извлечения 0 белых 0.375


Рассчитаем количество комбинаций белых мячей во 2м ящике из 5 по 4:

In [46]:
comb4_5 = comb(5, 4)
comb3_5= comb(5, 3)
comb2_5 = comb(5, 2)
comb1_5 = comb(5, 1)
print(f' количество комбинаций 4 белых {comb4_5} \n количество комбинаций 3 белых {comb3_5} \n количество комбинаций 2 белых {comb2_5} \n количество комбинаций 1 белого {comb1_5}')

 количество комбинаций 4 белых 5 
 количество комбинаций 3 белых 10 
 количество комбинаций 2 белых 10 
 количество комбинаций 1 белого 5


Рассчитаем количество комбинаций белых мячей во 2м ящике из 12 по 4:

In [48]:
comb4_12 = comb(12, 4)
comb3_12= comb(12, 3)
comb2_12 = comb(12, 2)
comb1_12 = comb(12, 1)
print(f' количество комбинаций 4 белых {comb4_12} \n количество комбинаций 3 белых {comb3_12} \n количество комбинаций 2 белых {comb2_12} \n количество комбинаций 1 белого {comb1_12}')

 количество комбинаций 4 белых 495 
 количество комбинаций 3 белых 220 
 количество комбинаций 2 белых 66 
 количество комбинаций 1 белого 12


Рассчитаем вероятности для второго ящика:

In [54]:
ver4_2 = ver(comb4_5, comb4_12)
ver3_2 = ver(comb3_5, comb3_12)
ver2_2 = ver(comb2_5, comb2_12)
ver1_2 = ver(comb1_5, comb1_12)
ver0_2 = 1 - ver1_2
print(f' вероятность извлечения 4 белых {ver4_2} \n вероятность извлечения 3 белых {ver3_2} \n вероятность извлечения 2 белых {ver2_2} \n вероятность извлечения 1 белого {ver1_2} \n вероятность извлечения 0 белых {ver0_2}')

 вероятность извлечения 4 белых 0.010101010101010102 
 вероятность извлечения 3 белых 0.045454545454545456 
 вероятность извлечения 2 белых 0.15151515151515152 
 вероятность извлечения 1 белого 0.4166666666666667 
 вероятность извлечения 0 белых 0.5833333333333333


Рассчитаем возможные комбинации из двух ящиков:

In [55]:
first_0second_4 = ver0_1 * ver4_2
first_0second_3 = ver0_1 * ver3_2
first_1second_2 = ver1_1 * ver2_2
first_2second_1 = ver2_1 * ver1_2

In [56]:
verall = first_0second_4 + first_0second_3 + first_1second_2 + first_2second_1
verall

0.2643398268398268

#### Ответ:

Вероятность того, что 3 мяча белые 26.4%

### Задание 3.

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

#### Решение:

Используем формулу Байеса и формулу полной вероятности:
$$P(B|A)=\frac{P(B) \cdot P(A|B)}{P(A)}$$
$$P(A)=P(B_1) \cdot P(A|B_1) + P(B_2) \cdot P(A|B_2) + ... +  P(B_n) \cdot P(A|B_n)$$


$P(A) = \frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6 = \frac{2.3}{3} \approx 0.77$ 

In [13]:
(0.9+0.8+0.6)/3

0.7666666666666667

a). $P(B_1|A)=\frac{\frac{1}{3} \cdot 0.9}{0.77} = \frac{0.3}{0.77} \approx 0.389 \; (38.9\%)$

In [17]:
0.3/0.77

0.3896103896103896

б). $P(B_2|A)=\frac{\frac{1}{3} \cdot 0.8}{0.77} \approx 0.346 \; (34.6\%)$

In [62]:
(0.8/3)/0.77

0.3463203463203463

d). $P(B_3|A)=\frac{\frac{1}{3} \cdot 0.6}{0.77} = \frac{0.2}{0.77} \approx 0.259 \; (25.9\%)$

In [63]:
0.2/0.77

0.25974025974025977

#### Ответ:

a) $\approx 39\%$ б) $\approx 35\%$ в) $\approx 26\%$

### Задание 4.

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

#### Решение:

Используем формулу Байеса и формулу полной вероятности:
$$P(B|A)=\frac{P(B) \cdot P(A|B)}{P(A)}$$
$$P(A)=P(B_1) \cdot P(A|B_1) + P(B_2) \cdot P(A|B_2) + ... +  P(B_n) \cdot P(A|B_n)$$

$A+B+C = \frac{1}{4} + \frac{1}{4} + \frac{1}{2} = 1$


$P(A) = \frac{1}{4} \cdot 0.8 + \frac{1}{4} \cdot 0.7 + \frac{1}{2} \cdot 0.9 = 0.2 + 0.175 + 0.45 = 0.825$ 

a). $P(B_1|A)=\frac{\frac{1}{4} \cdot 0.8}{0.825} = \frac{0.2}{0.825} \approx 0.242 \; (24.2\%)$

In [65]:
(0.8/4)/0.825

0.24242424242424246

б). $P(B_2|A)=\frac{\frac{1}{4} \cdot 0.7}{0.825} \approx 0.212 \; (21.2\%)$

In [66]:
(0.7/4)/0.825

0.21212121212121213

в). $P(B_3|A)=\frac{\frac{1}{2} \cdot 0.9}{0.825} = \frac{0.45}{0.825} \approx 0.545 \; (54.5\%)$

In [67]:
(0.9/2)/0.825

0.5454545454545455

#### Ответ:

a) $\approx 24.2\%$ б) $\approx 21.2\%$ в) $\approx 54.5\%$

### Задание 5.

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

#### Решение:

Используем формулу Байеса и формулу полной вероятности:
$$P(B|A)=\frac{P(B) \cdot P(A|B)}{P(A)}$$
$$P(A)=P(B_1) \cdot P(A|B_1) + P(B_2) \cdot P(A|B_2) + ... +  P(B_n) \cdot P(A|B_n)$$


$P(A) = \frac{1}{3} \cdot 0.1 + \frac{1}{3} \cdot 0.2 + \frac{1}{3} \cdot 0.25 = \frac{0.55}{3} \approx 0.183$ 

In [68]:
0.55/3

0.18333333333333335

a). Вероятность, что все детали выйдут из строя будет равна $P(B_1|A) \cdot P(B_2|A) \cdot P(B_3|A)$

$P(B_1|A)=\frac{\frac{1}{3} \cdot 0.1}{0.183} \approx 0.182 \; (18.2\%)$

In [69]:
(1/3 * 0.1)/0.183

0.18214936247723132

$P(B_2|A)=\frac{\frac{1}{3} \cdot 0.2}{0.183} \approx 0.364 \; (36.4\%)$

In [70]:
(1/3 * 0.2)/0.183

0.36429872495446264

$P(B_3|A)=\frac{\frac{1}{3} \cdot 0.25}{0.183} \approx 0.455 \; (45.5\%)$

In [71]:
(1/3 * 0.25)/0.183

0.4553734061930783

$P(B_1|A) \cdot P(B_2|A) \cdot P(B_3|A) = 0.182 \cdot 0.364 \cdot 0.455 \approx 0.030 \; (3.0\%)$

In [72]:
0.182 * 0.364 * 0.455

0.03014284

б). Вероятность, что только две детали выйдут из строя будет равна $P(B_1|A) + P(B_2|A) + P(B_3|A) + P(B_1|A) \cdot P(B_2|A) + (B_1|A) \cdot P(B_3|A) + (B_2|A) \cdot P(B_3|A) \approx 0.314 \; (31.4\%)$

In [76]:
0.182 * 0.364 + 0.182 * 0.455 + 0.364 * 0.455

0.314678

в). Вероятность, что хотя бы одна деталь выйдет из строя будет равна $1-P(B_0|A)$

$P(A) = 1 - 0.183 = 0.817$

$P(B_0|A)=\frac{\frac{1}{4} \cdot 0.45}{0.817} \approx 0.137 \; (13.7\%)$

In [80]:
(1/4 * 0.45)/0.817

0.13769889840881275

$1-P(B_0|A) = 1 - 0.137 = 0.863 \; (86.3\%)$

г). Вероятность, что от одной до двух деталей выйдут из строя будет равна $(P(B_1|A) + P(B_2|A) + P(B_3|A)) \cdot (P(B_1|A) \cdot P(B_2|A) + (B_1|A) \cdot P(B_3|A) + (B_2|A) \cdot P(B_3|A)) \approx 0.315 \; (31.5\%)$

In [79]:
(0.182 + 0.364 + 0.455) * (0.182 * 0.364 + 0.182 * 0.455 + 0.364 * 0.455)

0.314992678

#### Ответ:

a) $\approx 3.0\%$ б) $\approx 31.4\%$ в) $\approx 86.3\%$ г) $\approx 31.5\%$